components/status_dropdown/status_dropdown.jsx

```diff
@@ -115,6 +193,24 @@ export default class StatusDropdown extends React.PureComponent {
         const setDnd = needsConfirm ? () => this.showStatusChangeConfirmation('dnd') : this.setDnd;
         const setAway = needsConfirm ? () => this.showStatusChangeConfirmation('away') : this.setAway;
         const setOffline = needsConfirm ? () => this.showStatusChangeConfirmation('offline') : this.setOffline;
+        const setCustomTimedDnd = needsConfirm ? () => this.showStatusChangeConfirmation('dnd') : this.setCustomTimedDnd;
+
+        const dndSubMenuItems = this.dndTimes.map((time, index) => {
+            if (index === 5) {
+                return {
+                    id: `dndTime-${time}`,
```

Comment:
time has spaces and things like that. I would suggest to convert `time` to some kind of ids, and then use that id without spaces here. And also for the `text` that should be translated.

Reply:
how about splitting time at space and only using first element and also adding index of array?